src/backends/beamcoder.ts

```diff
@@ -306,13 +316,14 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
                 VERBOSE && console.log('returning previously filtered frame with pts', (closestFrame as Frame).pts);
                 closestFramePTS = (closestFrame as Frame).pts;
                 outputFrame = closestFrame;
+                this.#previousTargetPTS = targetPTS;
```

Comment:
I'm curious if we should set this at line 303 also.

Reply:
Also, looks like we already set it before returning. Except in one place (line 347).

Maybe we should add this before line 347 instead. I guess the goal is to make sure that this is set before we return from the function.